### ISSUE TITLE
Had to use next-transpile-modules to work with Next.js 12 (next@^12.0.2)

### ISSUE THREAD

**Issue by santanuyogi** [ndom91/react-timezone-select#46]
Thank you so much for your time on this. I want to mention that I spent quite some time today trying to get this to work with Next.js 12, and I kept running into these two errors:

1. `Cannot find module '/usr/src/app/node_modules/react/jsx-runtime`
2. `Module not found: Can't resolve 'react-timezone-select'`

The app would work fine without react-timezone-select installed, but as soon as I would install it, one of those two errors would pop up. I tried many different configurations of webpack, babel, next.config.js. Nothing worked until I finally tried adding the fix you mention with [next-transpile-modules](https://github.com/martpie/next-transpile-modules)...
```js
// next.config.js
const withTM = require("next-transpile-modules")(["react-timezone-select"])

module.exports = withTM({
  // ...further Next.js config
})
```
This is the only thing that worked for me using Next.js 12. If anyone else runs across this problem, maybe this post will be helpful.

package.json:
```
"next": "^12.0.2",
"next-transpile-modules": "^9.0.0",
"react": "^17.0.2",
"react-dom": "^17.0.2",
"react-select": "^5.1.0",
"react-timezone-select": "^1.1.13"
```

**Comment by ndom91** [comment]
Hi @santanuyogi, I hate to be the bearer of bad news, but this is mentioned directly in the readme :see_no_evil: haha. 

Unfortunately there still seems to be some sort of issue wiht Next.js 12 and this ESM package. I'm not sure if its something I'm doing wrong or something thats not supported on their end yet potentially. Haven't had the time to really look into it deeply.

**Comment by santanuyogi** [comment]
Maybe for now it would be best to update the [README.md](https://github.com/ndom91/react-timezone-select#readme) and remove the message:
```
Update: Next.js 12 now supports ESM packages by default. If you have any problems with this 
package in a Next.js 12 project, please let me know!
```
What do you think? Would you like me to edit and make a pull request?

**Comment by ndom91** [comment]
> Maybe for now it would be best to update the [README.md](https://github.com/ndom91/react-timezone-select#readme) and remove the message:
> 
> ```
> Update: Next.js 12 now supports ESM packages by default. If you have any problems with this 
> package in a Next.js 12 project, please let me know!
> ```
> 
> What do you think? Would you like me to edit and make a pull request?

I plan to look into this this weekend, if you really want I won't stop you making a PR, but it should hopefully be cleared up soon enough :smile:

**Comment by santanuyogi** [comment]
Ahh... sounds good. Thank you for being active on the project. If I run across anything that may help, I'll let you know.

**Comment by ndom91** [comment]
@santanuyogi okay so oddly enough, I just created a brand new next.js 12 project with the `create-next-app` defaults. It installed with `v12.0.3`. Then I installed `react-timezone-select` and added it to the index.js page and it just worked :thinking:

**Comment by alpritt** [comment]
I ran into a similar problem with running Jest tests. I added react-time-select to the jest `"transformIgnorePatterns"` setting in my project package.json, but I also had to add a `"main": "./dist/index.js"` to the react-time-select package.json to get the import to work. Not sure if that can be added or if it causes more problems. Or perhaps it would resolve the similar error with Next?

I think this issue is related: https://github.com/facebook/jest/issues/9771 which I found via https://github.com/facebook/jest/issues/11373 which is explaining the problem I was having but with a different project.
